openwisp_monitoring/device/tests/test_models.py

```diff
@@ -578,6 +578,90 @@ def _create_env(self):
         )
         return dm, ping, load, process_count
 
+    def test_disabling_critical_check(self):
+
+        Check = load_model('check', 'Check')
+        dm, ping, load, process_count = self._create_env()
+
+        ping_check_instance = Check.objects.create(
+            name='Ping Check',
+            check_type='ping',
+            content_object=dm.device,
+            params={},
+        )
+        dm.update_status('ok')
+
+        self.assertEqual(dm.status, 'ok')
+        with catch_signal(health_status_changed) as handler:
+            ping_check_instance.is_active = False
+            ping_check_instance.save()
+        if handler.call_count > 0:
+            call_args = handler.call_args[1]
+            self.assertEqual(call_args['instance'], dm)
+            self.assertEqual(call_args['status'], 'unknown')
+        dm.refresh_from_db()
+        self.assertEqual(dm.status, 'unknown')
+        dm.update_status('ok')
+
+    def test_saving_active_critical_check(self):
```

Comment:
I tried running this test on master and is not failing.
What are we trying to test here? Are we checking whether the new code affects normal operations?
In that case I think we can merge it with the previous tests: at the end of the previous test, just change any detail of the check, eg: the "name" and save again, then ensure the status hasn't changed and the signal hasn't been fired. This can be executed as a `subTest`.